lib/security/WSSecurityCert.js

```diff
@@ -43,6 +44,9 @@ function WSSecurityCert(privatePEM, publicP12PEM, password) {
     passphrase: password
   };
   this.x509Id = "x509-" + generateId();
+  this.hasTimeStamp = options.hasTimeStamp || typeof options.hasTimeStamp === 'boolean' ? !!options.hasTimeStamp : true;
```

Comment:
why do we need the `||`? wouldn't it be the same, if we'd just take the right side of the `or`?

Reply:
Ah, ok. 
To make those situations work, the left side of the check is needed, however it should be better defined, because:
```js
const options = {
  hasTimeStamp: 0
};
_hasTimeStamp = options.hasTimeStamp || typeof options.hasTimeStamp === 'boolean' ? !!options.hasTimeStamp : true; 
console.log(_hasTimeStamp); // true
```
which should be false.

```js
_hasTimeStamp = options.hasTimeStamp !== 'undefined' ? !!options.hasTimeStamp : true;
``` 
should work ...

Reply:
The following snippet will do the trick, transforming falsy/truthy values and leave `boolean` values untouched...

```javascript
const options = {
  hasTimestamp: 0
};

const hasTimestamp = !!options.hasTimestamp;
```

Reply:
Ok, changed it.